include/matx/core/tensor_impl.h

```diff
@@ -1094,6 +1165,9 @@ IGNORE_WARNING_POP_GCC
   protected:
     T *ldata_;
     Desc desc_;
+
+  public:
+    mutable std::optional<stf_logicaldata_type > *stf_ldata_;
```

Comment:
As discussed before this won't work since we can't use std:: objects on the device. It might work with `cuda::std::optional`, but we'd likely need to justify the overhead vs other options